networking/src/main/java/ar/com/wolox/wolmo/networking/retrofit/RetrofitServices.java

```diff
@@ -0,0 +1,143 @@
+     * and Retrofit.
+     * Please note that calling this method on an already initialized class will reset it to a
+     * clean state, configuring Retrofit to work with the endpoint provided in getApiEndpoint()
+     */
+    public void init() {
+        mServices = new HashMap<>();
+        mRetrofit = new Retrofit.Builder()
+                .baseUrl(getApiEndpoint())
+                .addConverterFactory(GsonConverterFactory.create(getGson()))
+                .client(getOkHttpClient())
+                .build();
+        mIsInitialized = true;
+    }
+
+    /**
+     * Returns the API endpoint.
+     *
+     * @return URL endpoint
+     */
+    @NonNull
+    public abstract String getApiEndpoint();
+
+    /**
+     * Returns an instance of Gson to use for conversion.
+     * This method calls <i>initGson(builder)</i> to configure the Gson Builder.
+     *
+     * @return A configured Gson instance
+     */
+    @NonNull
+    protected Gson getGson() {
+        com.google.gson.GsonBuilder builder = BaseGsonBuilder.getBaseGsonBuilder();
+        initGson(builder);
+        return builder.create();
+    }
+
+    /**
+     * Override if needed to configure a gson builder.
+     * You should add serializers and/or deserializers inside this method.
+     *
+     * @param builder Builder to configure
+     */
+    protected void initGson(@NonNull com.google.gson.GsonBuilder builder) {
+    }
+
+    /**
+     * Returns an OkHttpClient.
+     * This method calls <i>initClient(builder)</i> to configure the builder for OkHttpClient.
+     *
+     * @return A configured instance of OkHttpClient.
+     */
+    @NonNull
+    protected OkHttpClient getOkHttpClient() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        initClient(builder);
+        return builder.build();
+    }
+
+    /**
+     * Configures an <i>OkHttpClient.Builder</i>.
+     * You must add interceptors and configure the builder inside this method.
+     */
+    protected void initClient(@NonNull OkHttpClient.Builder builder) {
+        HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
+        loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+
+        builder.addInterceptor(loggerInterceptor);
+    }
+
+    /**
+     * Builds and returns a Retrofit Service.
+     * If the service wasn't accessed, it'll be created and cached internally.
+     * On successive requests, the already created instance will be returned.
+     * <p>
+     * Usage:
+     * Override this class and define the services like this:
+     * public static UserService user() {
+     * return getService(UserService.class);
+     * }
+     *
+     * @param clazz RetrofitService Class
+     * @param <T>   Service class
+     * @return service
+     */
+    public <T> T getService(@NonNull Class<T> clazz) {
+        if (!mIsInitialized) throw new RuntimeException("RetrofitServices is not initialized! " +
```

Comment:
Instead of using a `boolean` we can just check if `mRetrofit == null` for this and delete `mIsInitialized`. If you want to make it clearer, you can add a `isInitialized` `private method` that returns `mRetrofit == null`.